### PR TITLE
Kelsha Darby - Fixes - User Story

### LINKED ISSUE #129: User Story 44
Merchant User Can Delete An Item

Things to fix
- User should stay on "My Items" for the whole deletion process
- There is a delete button on the items show page

Whats Currently Happening
    - I go to My Items page
        - Delete button is present on every item (as it should be)
        - This link redirects to the items show page (the item is still present)
        - There is a delete link here too
        - When I click this link I am redirected to the "All Items" page (I should be on "My Items" for the whole deletion process & after)
        - There is also no flash message

### PR DESCRIPTION
**Pull request by kelshadarby**
## Merchant Item Deletion
Closes #129 

**_Test_**
- Add to merchant items index spec to test that when the delete button is hit it deletes the item and redirects to that merchants "My Items" page

**_Partial_**
- Remove the iteration from the partial

**_Views_**
- Added the first line of the iteration and the first section to each item index view and passed in the item found in the iteration to the partial

**_Routes_**
- Added a namespaced route for merchant item deletion
- Added a namespaced route for admin merchant item deletion

### CODE REVIEW

File: app/controllers/admin/merchant/items_controller.rb [hunk 1]
```diff
@@ -15,7 +15,14 @@ def update
...
 
-    def item_params
+  def destroy
+    item = Item.find(params[:id])
+    Review.where(item_id: item.id).destroy_all
+    item.destroy
+    redirect_to "/items"
+  end
+
+  def item_params
```
Comment (brian-greeson) [inline app/controllers/admin/merchant/items_controller.rb:25]: This is much better here

File: app/controllers/merchant/items_controller.rb [hunk 2]
```diff
@@ -31,6 +31,13 @@ def update
     redirect_to merchant_items_path
   end
 
+  def destroy
+    item = Item.find(params[:id])
+    Review.where(item_id: item.id).destroy_all
+    item.destroy
+    redirect_to merchant_items_path
+  end
+
```
Comment (brian-greeson) [inline app/controllers/merchant/items_controller.rb:40]: 👍

File: app/views/merchant/items/index.html.erb [hunk 3]
```diff
@@ -1,2 +1,6 @@
-<%= render partial: '/partials/item_index' %>
+<section class="grid-container item-list">
+  <% @merchant.items.each do |item| %>
+    <%= render partial: '/partials/item_index', locals: {item: item, path: merchant_item_destroy_path(item) } %>
```
Comment (brian-greeson) [inline app/views/merchant/items/index.html.erb:3]: Nice use of a partial

**Review by brian-greeson — APPROVED** [review]
Great!

**Review by fredrondina — APPROVED** [review]
nice partial incorperation

**Pull request merged**